ffmpeg/encoder.c

```diff
@@ -5,30 +5,56 @@
 #include <libavfilter/buffersrc.h>
 #include <libavfilter/buffersink.h>
 
-static int add_video_stream(struct output_ctx *octx, struct input_ctx *ictx)
+static AVStream *add_stream_copy(struct output_ctx *octx, AVStream *ist)
 {
-  // video stream to muxer
   int ret = 0;
+  if (!ist) LPMS_ERR(add_copy_err, "Input stream for copy not available");
   AVStream *st = avformat_new_stream(octx->oc, NULL);
-  if (!st) LPMS_ERR(add_video_err, "Unable to alloc video stream");
+  if (!st) LPMS_ERR(add_copy_err, "Unable to alloc copy stream");
   octx->vi = st->index;
   if (octx->fps.den) st->avg_frame_rate = octx->fps;
-  else st->avg_frame_rate = ictx->ic->streams[ictx->vi]->r_frame_rate;
+  else st->avg_frame_rate = ist->r_frame_rate;
+  st->time_base = ist->time_base;
+  ret = avcodec_parameters_copy(st->codecpar, ist->codecpar);
+  if (ret < 0) LPMS_ERR(add_copy_err, "Error copying params from input stream");
+  // Sometimes the codec tag is wonky for some reason, so correct it
+  ret = av_codec_get_tag2(octx->oc->oformat->codec_tag, st->codecpar->codec_id, &st->codecpar->codec_tag);
```

Comment:
Is this required? I don't see see `ret` used anywhere? Do the codec tags serve any purpose here?

Reply:
Yeah, LPMS_ERR expects ret to be defined and will add to error description. Not that I love this design, mind you, but was told to proceed in small steps. If I were to rewrite the code from scratch it will be very different.

I have no idea about codec tags, but again, was told that compatibility with old code is a big concern. This is what old code was doing every time, and so does mine.